git_dumper.py

```diff
@@ -459,10 +478,18 @@ def fetch_git(url, directory, jobs, retry, timeout, http_headers, client_cert_p1
             jobs,
             args=(url, directory, retry, timeout, http_headers),
         )
+        
+        os.chdir(directory)
+
+        printf("[-] Sanitizing .git/config\n")
+        sanitize_file(".git/config")
 
         printf("[-] Running git checkout .\n")
-        os.chdir(directory)
-        subprocess.check_call(["git", "checkout", "."])
+        ENV = os.environ
```

Comment:
I think we shouldn't avoid modifying the current environment in case someone uses git-dumper as a library (although unlikely). We can simply make a copy.
Also, nitpicking but I would just call it "environment". ENV makes it seem like it's a global variable, which it isn't.

```suggestion
        environment = os.environ.copy()
```

Reply:
if so, `git checkout` should not be called at all as someone might assume that the whole traffic is transported over configured proxy and I think subprocess will not use monkeypatched sockets. Perhaps adding optional flag `--git-checkout` with a proper explanation would suffice.

Reply:
Sorry, I think there is a misunderstanding.
We should definitely call git checkout . with the modified environment. I'm just saying we shouldn't modify `os.environ` in case it's used somewhere else. TL, DR: Just add a `.copy()` to `os.environ`.